api/src/test/java/org/apache/iceberg/types/TestTypes.java

```diff
@@ -90,5 +90,42 @@ public void testNestedFieldBuilderIdCheck() {
     assertThatExceptionOfType(NullPointerException.class)
         .isThrownBy(() -> required("field").ofType(Types.StringType.get()).build())
         .withMessage("Id cannot be null");
+
+    assertThat(Types.fromPrimitiveString("geometry")).isEqualTo(Types.GeometryType.get());
```

Comment:
this should be in its own test

Reply:
In this suite, these cases would be added to `fromPrimitiveString` and `fromTypeName`.

Reply:
Moved tests to `fromTypeName` and `fromPrimitiveString`.